lib/pbench/agent/results.py

```diff
@@ -102,100 +122,78 @@ def make_result_tb(self):
+            RestAPI PUT request
+    """
+
     chunk_size = 4096
 
-    def __init__(self, controller, tarball, config, logger):
+    def __init__(
+        self, controller: str, tarball: str, config: PbenchAgentConfig, logger: Logger
+    ):
         self.tarball = Path(tarball)
         if not self.tarball.exists():
-            logger.error("tarball does not exist, '{}'", tarball)
-            sys.exit(1)
-        self.tarball_md5 = Path(f"{tarball}.md5")
-        if not self.tarball_md5.exists():
-            logger.error("tarball's .md5 does not exist, '{}'", self.tarball_md5)
-            sys.exit(1)
+            raise FileNotFoundError(f"Tarball does not exist, '{self.tarball}'")
         self.logger = logger
-        server_rest_url = config.results.get("server_rest_url")
+        server_rest_url = config.get("results", "server_rest_url")
         self.upload_url = f"{server_rest_url}/upload/ctrl/{controller}"
 
-    def read_in_chunks(self, file_object):
+    def read_in_chunks(self, file_object: IO) -> Iterator[AnyStr]:
         data = file_object.read(self.chunk_size)
         while data:
             yield data
             data = file_object.read(self.chunk_size)
 
-    def copy_result_tb(self):
-        files = [f for f in self.tarball.parent.iterdir() if f.is_file()]
-        file_count = len(files)
-        if file_count != 2:
-            self.logger.error(
-                "(internal): unexpected file count, {}, associated with tarball, '{}'",
-                file_count,
-                self.tarball,
-            )
-            sys.exit(1)
-
-        with open(self.tarball_md5, "r") as md5fp:
-            md5sum = md5fp.read().split()[0]
-        filename = secure_filename(str(self.tarball))
-        headers = {"filename": filename, "Content-MD5": md5sum}
+    def copy_result_tb(self, token: str) -> None:
+        """copy_result_tb - copies tb from agent to configured server upload URL
+
+            Args
+                token -- an Authorization key which will allow the user to be
+                    authorized by the server.
+        """
+        headers = {
+            "filename": secure_filename(str(self.tarball)),
+            "Content-MD5": md5sum(self.tarball),
+            "Authorization": f"Bearer {token}",
+        }
         with self.tarball.open("rb") as f:
             try:
                 response = requests.put(
                     self.upload_url, data=self.read_in_chunks(f), headers=headers
                 )
                 self.logger.info("File uploaded successfully")
-            except Exception:
-                self.logger.exception("There was something wrong with your request")
-                sys.exit(1)
-        if not response.status_code == 200:
-            self.logger.error(
-                "There was something wrong with your request, error code: '{}'",
-                response.status_code,
+            except requests.exceptions.ConnectionError:
```

Comment:
Hmm, I didn't notice this before; but without `response.raise_for_status()` we shouldn't actually be getting exceptions from the `put`.... Are these being tested?

Reply:
Since we're not using `raise_for_status()`, HTTP errors (unlike network errors like `ConnectionError`) currently fall through to line 195 where they are handled.

Raising them here instead would allow me to get rid of that seemingly redundant error at line 196, so I'll give that a try.

As for testing, I believe we test `ConnectionError`'s but not HTTP errors.  Got suggestions on which ones to generate?

Reply:
It seems good.  I added a test for a `404` response.  Let me know if I should change that to something more plausible, or if there are others that you think I should add.